x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/matrix_histogram/helpers.ts

```diff
@@ -18,7 +18,8 @@ export const getGenericData = <T>(
 ): MatrixHistogramData[] => {
   let result: MatrixHistogramData[] = [];
   data.forEach((bucketData: unknown) => {
-    const group = get('key', bucketData);
+    // if key is not a string, use key_as_string instead.
```

Comment:
Isn't this if `key_as_string` is present use it, else default to the existing key?